src/lib.rs

```diff
@@ -478,3 +502,16 @@ macro_rules! multiaddr {
         }
     }
 }
+
+impl From<SocketAddr> for Multiaddr {
+    fn from(socket: SocketAddr) -> Self {
+        match socket {
+            SocketAddr::V4(sock) => Self::empty()
+                .with(Protocol::Ip4(*sock.ip()))
+                .with(Protocol::Tcp(sock.port())),
```

Comment:
`SocketAddr` might as well specify a UDP address, thus this `From` implementation is not correct.